packages/gotrue/lib/src/types/mfa.dart

```diff
@@ -257,7 +257,24 @@ class AuthMFAGetAuthenticatorAssuranceLevelResponse {
   });
 }
 
-enum AMRMethod { password, otp, oauth, totp, magiclink }
+enum AMRMethod {
+  password('password'),
+  otp('otp'),
+  oauth('oauth'),
+  totp('totp'),
+  magiclink('magiclink'),
+  recovery('recovery'),
+  invite('invite'),
+  ssoSaml('sso/saml'),
+  emailSignUp('email/signup'),
+  emailChange('email_change'),
+  tokenRefresh('token_refresh'),
+  anonymous('anonymous'),
+  mfaPhone('mfa/phone');
+
```

Comment:
Another thing mentioned from the auth team was there might be others added in the future. Could we have a `unknown` or something to handle values that gets added in the future versions?